File: sdk/java/src/test/java/com/flagent/client/model/DistributionRequestTest.java
package com.flagent.client.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for DistributionRequest
 */
class DistributionRequestTest {

    @Test
    void testDistributionRequest() {
        DistributionRequest model = new DistributionRequest()
            .variantID(1L)
            .variantKey("control")
            .percent(100L);
        Assertions.assertNotNull(model);
        Assertions.assertEquals(1L, model.getVariantID());
        Assertions.assertEquals("control", model.getVariantKey());
        Assertions.assertEquals(100L, model.getPercent());
    }

    @Test
    void variantIDTest() {
        DistributionRequest model = new DistributionRequest().variantID(5L);
        Assertions.assertEquals(5L, model.getVariantID());
    }

    @Test
    void variantKeyTest() {
        DistributionRequest model = new DistributionRequest().variantKey("variant_a");
        Assertions.assertEquals("variant_a", model.getVariantKey());
    }

    @Test
    void percentTest() {
        DistributionRequest model = new DistributionRequest().percent(50L);
        Assertions.assertEquals(50L, model.getPercent());
    }

}

File: sdk/dart/test/distribution_request_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for DistributionRequest
void main() {
  final instance = DistributionRequest((b) => b
    ..variantID = 1
    ..variantKey = 'control'
    ..percent = 50);

  group(DistributionRequest, () {
    test('to test the property `variantID`', () async {
      expect(instance.variantID, equals(1));
    });

    test('to test the property `variantKey`', () async {
      expect(instance.variantKey, equals('control'));
    });

    test('to test the property `percent`', () async {
      expect(instance.percent, equals(50));
    });
  });
}
